src/waddle/processing/segment.py

```diff
@@ -144,49 +146,46 @@ def process_segments(
     Args:
         segs_folder_path (str): Path to the folder containing the speech segments.
         combined_audio_path (str): Path to save the combined audio file.
-        transcription_output_path (str): Path to save the combined transcription file.
+        transcription_path (str): Path to save the combined transcription file.
         language (str): Language code for transcription.
     """
-    segs_folder_path = Path(segs_folder_path)  # TODO: Delete it after switch to Pathlib in test
-    combined_audio_path = Path(
-        combined_audio_path
-    )  # TODO: Delete it after switch to Pathlib in test
-    transcription_output_path = Path(
-        transcription_output_path
-    )  # TODO: Delete it after switch to Pathlib in test
+    if transcribe and transcription_path is None:
```

Comment:
if `transcription_path` is required for `transcribe == True`, I might get rid of `transcribe` and skip the process if `transcription_path is None`

Reply:
I think the purpose of default arguments is to enable calling the function without specifying all parameters. It feels strange that if I only provide the required arguments, I would end up in this branch.

Reply:
Indeed, that does seem better.